File: Programmers/Level2/카펫/카펫.py
def solution(brown, yellow):
    answer = []

    for yellow_y in range(1, int(yellow**(0.5)) + 1):
        if not (yellow % yellow_y):
            yellow_x = yellow // yellow_y

            if (2 * yellow_x) + (2 * yellow_y) + 4 == brown:
                answer = [yellow_x + 2, yellow_y + 2]

    return answer

brown = 24
yellow = 24

print(solution(brown, yellow))

File: Programmers/Level2/카펫/카펫.js
function solution(brown, red) {
  const total = brown + red;

  for (let i = Math.floor(total / 2); i > 0; i--) {
    if (total % i !== 0) continue;

    const width = i;
    const height = total / i;

    if ((width - 2) * (height - 2) === red) {
      return [width, height];
    }
  }
}

const brown = 24;
const yellow = 24;

console.log(solution(brown, yellow));
